sdk-extensions/baggage-processor/src/test/java/io/opentelemetry/sdk/extension/trace/baggage/TestBaggageSpanProcessor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.trace.baggage;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class TestBaggageSpanProcessor {
+  @Mock private SpanProcessor nextProcessor;
+  @Mock private ReadWriteSpan readWriteSpan;
+
+  @Test
+  void ignoresEmptyBaggage() {
+    BaggageSpanProcessor processor = BaggageSpanProcessor.builder(nextProcessor).build();
+
+    Baggage empty = Baggage.empty();
+    Context context = Context.root().with(empty);
+
+    processor.onStart(context, readWriteSpan);
+    Mockito.verify(nextProcessor).onStart(context, readWriteSpan);
+    Mockito.verify(readWriteSpan, Mockito.never())
+        .setAttribute(Mockito.anyString(), Mockito.anyString());
+  }
+
+  @Test
+  void appendsAllBaggage() {
+    BaggageSpanProcessor processor = BaggageSpanProcessor.builder(nextProcessor).build();
+    Baggage empty = Baggage.builder().put("test1", "value").put("test2", "value").build();
+    Context context = Context.root().with(empty);
+    processor.onStart(context, readWriteSpan);
+    Mockito.verify(nextProcessor).onStart(context, readWriteSpan);
+    Mockito.verify(readWriteSpan).setAttribute("test1", "value");
```

Comment:
Consider replacing `verify` with `when` and returning the `readWriteSpan`. Mockito now automatically verifies when using `when` so don't need two statements. And it ensures the business logic can be changed to chain invocations if needed in the future (I guess it might not happen, but it prevents the possibility of a really mysterious NPE in the future if it does).

I believe the `never` calls aren't really needed either (maybe they're good for documentation though) since mockito would fail if there are unexpected invocations too

Reply:
This depends on whether someone is using strict-mock or lenient-mock mode.  I've learned to be a bit more explicit in negative assumptions (never) in case someone flips to lenient mode.

IF you think the `when` calls read better, I can flip this, I just felt the post-call verification felt more natural here.

Reply:
It's two types of fragility, the other being methods that never return null returning null in the mocked version. I think we can expect to always be in the default strict mock mode.